keanu-project/src/main/java/io/improbable/keanu/network/LambdaSection.java

```diff
@@ -90,18 +90,18 @@ public static LambdaSection getDownstreamLambdaSection(Vertex<?> aVertex, boolea
      * @param shouldAdd    true when a give vertex should be included in the result false otherwise
      * @return A Set of vertices that are in the direction implied by nextVertices and filtered by shouldAdd
      */
-    private static Set<Vertex> getVerticesDepthFirst(Vertex vertex,
-                                                     Function<Vertex, Collection<Vertex>> nextVertices,
-                                                     Predicate<Vertex> shouldAdd) {
+    public static Set<Vertex> getVerticesDepthFirst(Vertex vertex,
```

Comment:
why the change to `public`? Can be package private if we're just making it public to access it in tests

Reply:
I'll need it to be public in a auto diff refactor I'm working on. Plus it's a pure static function so there's no harm in exposing it. Plus the package private for testing is a bit cheeky.

Reply:
ok